HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt

```diff
@@ -264,8 +261,11 @@ class HyprMXAdapter : PartnerAdapter {
      * @param consentStatus the consent status value to be stored.
      */
     private fun setUserConsent(context: Context, consentStatus: ConsentStatus) {
-        context.getSharedPreferences(HYPRMX_PREFS_KEY, Context.MODE_PRIVATE).edit()
-            .putInt(HYPRMX_USER_CONSENT_KEY, consentStatus.ordinal).apply()
+        context.getSharedPreferences(HYPRMX_PREFS_KEY, Context.MODE_PRIVATE)
+            .edit()
+            .putInt(HYPRMX_USER_CONSENT_KEY, consentStatus.ordinal)
+            .apply()
+
         HyprMX.setConsentStatus(consentStatus)
     }
```

Comment:
not something immediately but to be safe here and elsewhere I'd recommend `.commit()` over `.apply()` since we will know the result of the write before we decide if we should call `HyprMX.setConsentStatus(consentStatus)`. otherwise like right now the two things will just happen independently of each other—you could end up with a failed SharedPreferences write but still call `HyprMX.setConsentStatus(consentStatus)`. and when you get the value back from the SP later it doesn't exist so it'll be confusing.

something like this but you can simplify it:

```kt
private fun setUserConsent(context: Context, consentStatus: ConsentStatus) {
    val prefsWriteSucceeded = context.getSharedPreferences(HYPRMX_PREFS_KEY, Context.MODE_PRIVATE)
        .edit()
        .putInt(HYPRMX_USER_CONSENT_KEY, consentStatus.ordinal)
        .commit()

    when(prefsWriteSucceeded) {
        true -> {
            HyprMX.setConsentStatus(consentStatus)
        }
        else -> {
            // Log or handle the error
        }
    }
}
```

but, that's a really edge case 😛

Reply:
Sure. Let's aim to be safer. 🗒️ 👍🏼

Reply:
doesn't strict mode flag for using commit on the main thread? It's probably fine since it's one int. either way, it's probably fine.
oh, you're in a suspend function already. you could make this a suspend function and boop this over to the IO thread and then commit() is fine.

Reply:
Oh, yeah, I can turn them to a suspend function and do the commit() there. Let me do that. 🗒️ ✍🏼

Reply:
@bichenwang actually, should we still do the suspend thing:
<img width="338" alt="image" src="https://github.com/ChartBoost/chartboost-mediation-android-adapter-hyprmx/assets/30056771/db5ad584-b57c-4cf6-b265-1fffbd88b836">